paper/draft.md

```diff
@@ -0,0 +1,28 @@
+> Here we will put the text for the paper. Later, it will be slightly reformated to meet JOSS's submission criteria. For now, it's mostly copy-paste-edit from the manual or readme.
+
+
+## Schema Salad: A bridge between document and record oriented data modeling and the Semantic Web
+
+### Summary
+Salad is a schema language for describing structured linked data documents in JSON or YAML documents. A Salad schema provides rules for preprocessing, structural validation, and link checking for documents described by a Salad schema. Salad builds on JSON-LD and the Apache Avro data serialization system and extends Avro with features for rich data modeling such as inheritance, template specialization, object identifiers, and object references. Salad was developed to provide a bridge between the record oriented data modeling supported by Apache Avro and the Semantic Web.
+
+### Statement of need
+The JSON data model is a popular way to represent structured data. It is attractive because of its relative simplicity and is a natural fit with the standard types of many programming languages. However, this simplicity comes at the cost that basic JSON lacks expressive features useful for working with complex data structures and document formats, such as schemas, object references, and namespaces.
+
+JSON-LD is a W3C standard providing a way to describe how to interpret a JSON document as Linked Data by means of a "context". JSON-LD provides a powerful solution for representing object references and namespaces in JSON based on standard web URIs but is not itself a schema language. Without a schema providing a well-defined structure, it is difficult to process an arbitrary JSON-LD document as idiomatic JSON because there are many ways to express the same data that are logically equivalent but structurally distinct.
```

Comment:
It seems like we should add some final sentence at the end giving the idea "And that's why we developed schema-salad". Otherwise, the thoughts go from JSON to JSON-LD and its limitations and then breaks off.